src/test/scala/DateFormatTest.scala

```diff
@@ -6,36 +6,243 @@ class DateFormatTest extends FunSuite {
+    val maybeParsedDateTime = dateFormatObj.parse(
+      "99" // the century must not be truncated
+    )
+    assert(maybeParsedDateTime.isEmpty)
+  }
+
+  test("it should parse gYearMonth") {
+    val dateFormatObj =
+      DateFormat(None, "http://www.w3.org/2001/XMLSchema#gYearMonth")
+    val maybeParsedDateTime = dateFormatObj.parse(
+      "2004-04-05:00"
+    )
+    assert(maybeParsedDateTime.isDefined)
+    val parsedDateTime = maybeParsedDateTime.get
+    assert(parsedDateTime.getYear == 2004)
+    assert(parsedDateTime.getMonthValue == 4)
+    assert(parsedDateTime.getOffset.getTotalSeconds == -5 * 60 * 60)
+  }
+
+  test("it should not return gYearMonth for invalid values") {
+    val dateFormatObj =
+      DateFormat(None, "http://www.w3.org/2001/XMLSchema#gYearMonth")
+    val maybeParsedDateTime = dateFormatObj.parse(
+      "2004-4" //the month must be two digits
+    )
+    assert(maybeParsedDateTime.isEmpty)
+  }
+
+  test("it should parse time") {
+    val dateFormatObj =
+      DateFormat(None, "http://www.w3.org/2001/XMLSchema#time")
+    val maybeParsedDateTime = dateFormatObj.parse(
+      "13:20:30.5555-05:00"
+    )
+    assert(maybeParsedDateTime.isDefined)
+    val parsedDateTime = maybeParsedDateTime.get
+    assert(parsedDateTime.getOffset.getTotalSeconds == -5 * 60 * 60)
+    assert(parsedDateTime.getNano == 0.5555 * 1e9)
+    assert(parsedDateTime.getHour == 13)
+    assert(parsedDateTime.getMinute == 20)
+    assert(parsedDateTime.getSecond == 30)
+  }
+
+  test("it should not return time for invalid values") {
+    val dateFormatObj =
+      DateFormat(None, "http://www.w3.org/2001/XMLSchema#time")
+    val maybeParsedDateTime = dateFormatObj.parse(
+      "5:20:00" // hours, minutes, and seconds must be two digits each
+    )
+    assert(maybeParsedDateTime.isEmpty)
+  }
+
+  test("it should not return dateTimeStamp for invalid values") {
+    val dateFormatObj =
+      DateFormat(None, "http://www.w3.org/2001/XMLSchema#dateTimeStamp")
+    val maybeParsedDateTime = dateFormatObj.parse(
+      "2004-04-12T13:20:00"
+    )
+    assert(maybeParsedDateTime.isEmpty)
+  }
+
+  test("it should parse dateTimeStamp") {
+    val dateFormatObj =
+      DateFormat(None, "http://www.w3.org/2001/XMLSchema#dateTimeStamp")
+    val maybeParsedDateTime = dateFormatObj.parse(
+      "2004-04-12T13:20:00-05:00" // Will fail for this value - check this "2004-04-12T13:20:00-05:00" from http://www.datypic.com/sc/xsd11/t-xsd_dateTimeStamp.html
+    )
+    assert(maybeParsedDateTime.isDefined)
+    val parsedDateTime = maybeParsedDateTime.get
+    assert(parsedDateTime.getOffset.getTotalSeconds == -5 * 60 * 60)
+
+  }
+
+  test("it should parse date") {
+    val dateFormatObj =
+      DateFormat(None, "http://www.w3.org/2001/XMLSchema#date")
+    val maybeParsedDateTime = dateFormatObj.parse(
+      "-0045-01-01"
+    )
+    // edge case "-0045-01-01" is a valid date but it cant be parsed properly
```

Comment:
This comment seems a bit out of date if the test passes.

Reply:
Removed